src/pybind/mgr/dashboard/controllers/cluster_upgrade.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+
+from typing import Dict, List, Optional
+
+from ..services.exception import handle_orchestrator_error
+from ..services.orchestrator import OrchClient, OrchFeature
+from . import APIDoc, APIRouter, CreatePermission, Endpoint, EndpointDoc, \
+    ReadPermission, RESTController, UpdatePermission, allow_empty_body
+from .orchestrator import raise_if_no_orchestrator
+
+
+@APIRouter('/cluster/upgrade')
```

Comment:
And let's restrict it to the user who only has admin privileges 
```suggestion
@APIRouter('/cluster/upgrade', Scope.CONFIG_OPT)
```